### DISCUSSION TITLE
Tokio + Winit?

### DISCUSSION THREAD

**Discussion by FredrikNoren** [tokio-rs/tokio#2953] [Q&A] [answered]
Hi,

I'm trying to get [winit](https://github.com/rust-windowing/winit) to work with tokio, but the problem is that winit has it's own event loop which needs to run on the main thread. So the question is; how can I make these two work together?

I have a couple of ideas how to do this; one would be to run the tokio runtime on a separate thread and just let winit have the main thread. I was thinking I could create a channel to communicate messages from winit to my tokio code. But is it possible to run the tokio runtime in a separate thread?

Or are there other strategies that would work better?

Thanks!

**Comment by Darksonn** [comment, marked as answer]
You should run the Tokio runtime on a separate thread. You can do this by creating a new `tokio::runtime::Runtime` object and calling `spawn` on it to create the root future.

**Reply by FredrikNoren** [reply]
Great, thank you!
